driver.go

```diff
@@ -273,40 +245,21 @@ func (d *DockerOnTop) Unmount(request *volume.UnmountRequest) error {
 	}
 	defer activemountsdir.Close() // There's nothing I could do about the error if it occurs
 
-	dirEntries, readDirErr := activemountsdir.ReadDir(2) // Check if there is any _other_ container using the volume
-	if len(dirEntries) == 1 || errors.Is(readDirErr, io.EOF) {
-		// If just one entry or directory is empty, unmount overlay and clean up
-
+	doUnmountFs, err := d.deactivateVolume(request.Name, request.ID, activemountsdir)
+	if err != nil {
+		log.Errorf("Error while activating the filesystem mount: %w", err)
+		return internalError("failed to deactivate the active mount:", err)
+	} else if doUnmountFs {
```

Comment:
Oh-oh! From what I can see in the code of `.deactivateVolume`, it seems like you want this code to take actions for `doUnmountFs == true` even when `err != nil`! Am I right?

Reply:
You are right, I did not notice there was a case returning true + error. I have changed it to log a warning instead and return true + nil